static/app/views/releases/components/useFinalizeRelease.tsx

```diff
@@ -0,0 +1,32 @@
+import type {Release} from 'sentry/types/release';
+import {fetchMutation, useMutation} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type TPayload = {dateReleased: string};
+type TData = unknown;
+type TError = unknown;
+type TVariables = [release: Release];
+type TContext = unknown;
+
+export default function useFinalizeRelease() {
+  const organization = useOrganization();
+  const api = useApi({
+    persistInFlight: false,
+  });
+
+  return useMutation<TData, TError, TVariables, TContext>({
+    mutationFn: ([release]) => {
+      const payload: TPayload = {
+        dateReleased: release.firstEvent ?? release.dateCreated,
```

Comment:
Why `firstEvent` > `dateCreated`?

Also why not current time?

Reply:
I think current time is tooo late in many cases. It affects sort order, so date order matters too.

It's likely that there will either be CI setup to do this automatically, which would result in a fine ordering. But if people are manually clicking things in the UI then it's likely that they'll be clicking in any order (or maybe, top-to-bottom on the page) which will mess things up. I think that picking a reasonable early-date automatically makes sense in the ui and will let users manually bucket releases by finalized/un-finalized if they're not using the automation.

Reply:
can you cp this into a code comment?